Hackatosh/2019-soul-pet
# Feature/events

- Added custom validators in **utils.ts**
- Added _PetEvent_ model to store events in DB in **event.ts** (and modified **initDB.ts** and **associations.ts**)
- Deleted unused timestamp in models and fixed associations
- Created an events router in **events.ts** which allows to :
-> Create a new event
-> Get an event by eventId
-> Modifiy an event by eventId
->Delete an event by eventId
-> Add an animal to an event
-> Delete an animal from an event
Event can be modified only by the user who created them and can be then accessed by everyone logged in. An animal can be added and deleted only by its owner.

**To test :** 
- Register a user and log in
- Add an animal for this user
- Create an event (with _POST_ on _/events/_) and then access its informations (with _GET_ on _/events/:eventId_)
- Modify the event (with _PUT_ on _/events/:eventId_) and see the modifications (with _GET_ on _/events/:eventId_)
- Add an animal (with _POST_ on _/events/:eventId/animals_) and see the animal added (with _GET_ on _/events/:eventId_)
- Delete the animal (with _DELETE_ on _/events/:eventId/animals/:animalId_) and see that the animal has been deleted (with _GET_ on _/events/:eventId_)
- Delete the event (with _DELETE_ on _/events/:eventId_) and see that it has been deleted (with _GET_ on _/events/:eventId_)
